### PR TITLE
APIM -APRL recommendation

### PR DESCRIPTION
**Pull request by aarthiem**
# Overview/Summary

APIM -APRL recommendation  for stv2 APIM upgrades

## Related Issues/Work Items
https://dev.azure.com/CSUSolEng/Well-Architected%20Framework/_workitems/edit/29872

- To associate a GitHub Issue, use a [key word](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) preceded with the GitHub issue number.
- To associate an ADO Work Item, use the key word `AB#` succeeded with the [ADO Work Item ID](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## This PR fixes/adds/changes/removes

1. adds recommendation for stv2 APIM upgrades

### Breaking Changes

1. NA

## As part of this Pull Request I have

- [ x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library/contributing) and ensured this PR is compliant with the guide
- [x ] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library/pulls)
- [ x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library/issues) or ADO Work Items (Internal Only)
- [ x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library/tree/main)
- [x ] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries and/or scripts
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by ejhenry** [comment]
Thanks for the contribution @aarthiem! Are you able to work through the failed linting check or do you need help with that?

File: docs/content/services/integration/api-management/code/apim-3/apim-3.kql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Azure Resource Graph Query
+// Find all API Management instances that aren't upgraded to platform version stv2
+resources
+| where type =~ 'Microsoft.ApiManagement/service'
+| extend plat_version = properties.platformVersion
+| extend skuName = sku.name
+| where tolower(plat_version) != tolower('stv2')
+| project recommendationId = "apim-3", name, id, tags, param1=strcat("Platform Version: ", plat_version) , param2=strcat("SKU: ", skuName)
```
Comment (ejhenry) [inline docs/content/services/integration/api-management/code/apim-3/apim-3.kql:8]:
```suggestion
| project recommendationId = "apim-3", name, id, tags, param1=strcat("Platform Version: ", plat_version) , param2=strcat("SKU: ", skuName)

```

File: docs/content/services/integration/api-management/_index.md [hunk 2]
```diff
@@ -75,3 +75,28 @@ Enable zone redundancy for APIM instances. With zone redundancy, the gateway and
...
+
+**Resource Graph Query/Scripts**
+
+{{< collapse title="Show/Hide Query/Script" >}}
+
+{{< code lang="sql" file="code/apim-3/apim-3.kql" >}} {{< /code >}}
+
+{{< /collapse >}}
+
+<br><br>
```
Comment (ejhenry) [inline docs/content/services/integration/api-management/_index.md:102]:
```suggestion
<br><br>

```

**Comment by ejhenry** [comment]
@aarthiem - I resolved the linting issues. Can you include in the PR a screenshot of the successful testing of your ARG query?

File: docs/content/services/integration/api-management/_index.md [hunk 3]
```diff
@@ -75,3 +75,28 @@ Enable zone redundancy for APIM instances. With zone redundancy, the gateway and
 {{< /collapse >}}
 
 <br><br>
+
+### APIM-3 - Upgrade to platform version stv2
+
+**Category: Compliance**
```
Comment (ejhenry) [inline docs/content/services/integration/api-management/_index.md:81]:
```suggestion
**Category: Availability**
```

**Review by ejhenry — COMMENTED** [review]
Hi @aarthiem - Please update the summary table to include the new recommendation. 

I suggested a category from the allowed list in the contribution guide. If you think one of the other options would be a better fit, please change to that.

Also, please include a screenshot of the successful testing of your query.

File: docs/content/services/integration/api-management/_index.md (hunk 3 above)
Comment (aarthiem) [reply docs/content/services/integration/api-management/_index.md:81]: Hi @ejhenry - I do not have stv1 instances but stv2 only, the below screenshot shows the stv2 instances where I have changed the filter to look for text not with stv1. In my commit, I have looked for text not with stv2. 
image

File: docs/content/services/integration/api-management/_index.md
Comment (ejhenry) [inline docs/content/services/integration/api-management/_index.md:1]: @aarthiem please update the summary table at the top of this page to include your new recommendation. Thanks.

Comment (aarthiem) [reply docs/content/services/integration/api-management/_index.md:1]: @ejhenry - updated the table, please check

**Review by oZakari — APPROVED** [review]

**Review by oZakari — CHANGES_REQUESTED** [review]
Just waiting for screenshot as discussed, but otherwise looks good!

**Comment by aarthiem** [comment]
image
Screenshot 2024-04-04 192639
@ejhenry @edknox @oZakari - Please check the screenshot here.

**Review by oZakari — APPROVED** [review]

**Pull request merged**